lib/rules/sort-prop-types.js

```diff
@@ -57,7 +57,9 @@ module.exports = {
     const propWrapperFunctions = new Set(context.settings.propWrapperFunctions || []);
 
     function getKey(node) {
-      return sourceCode.getText(node.key || node.argument);
+      const key = sourceCode.getText(node.key || node.argument);
+      const hasLeadingQuote = /['"]/.test(key[0]);
+      return hasLeadingQuote ? key.substring(1, key.length - 1) : key;
```

Comment:
Yes, precisely that, but also because it doesn't support negative indices, and because using string slice has the same API as array slice, so it's more intuitive.